index.html

```diff
@@ -410,6 +463,94 @@ <h3 id="installation-sec">
       </h2>
       <section>
         <h3>
+          <code>BeforeInstallPromptEvent</code>
+        </h3>
+        <pre class="idl">
+enum AppBannerPromptOutcome {
+  "accepted",
+  "dismissed"
+};
+
+[Constructor(DOMString typeArg, optional BeforeInstallPromptEventInit eventInit)]
+interface BeforeInstallPromptEvent : Event {
+    readonly attribute Promise&lt;AppBannerPromptOutcome&gt; userChoice;
```

Comment:
I find the fact that this is a promise attribute (as opposed to a function that returns a promise) really weird. I know it's what Chrome does. I wonder if it's too late to change it?

It causes a lot of weirdness because there are side-effects upon reading the attribute. For example, in Chrome a user-constructed `BeforeInstallPromptEvent` logs a console error just by the event object itself being printed to the console, because it tries to read the `userChoice` attribute. And [this](https://bugs.chromium.org/p/chromium/issues/detail?id=655877).